backend/src/main/resources/org/cryptomator/hub/flyway/V4__Update_PKs.sql

```diff
@@ -0,0 +1,50 @@
+-- noinspection SqlNoDataSourceInspectionForFile
+
+-- temporarily drop views
+DROP VIEW "effective_vault_access";
+DROP VIEW "effective_group_membership";
+
+-- temporarily disable foreign key constraints
+ALTER TABLE "vault_access" DROP CONSTRAINT "VAULT_ACCESS_FK_VAULT";
+ALTER TABLE "access_token" DROP CONSTRAINT "ACCESS_FK_VAULT";
+ALTER TABLE "access_token" DROP CONSTRAINT "ACCESS_FK_DEVICE";
+
+-- collate C is more efficient on indexed columns (sadly user ids are generated externally and there is no guarantee for them to be UUIDs):
+ALTER TABLE "authority" ALTER COLUMN "id" SET DATA TYPE VARCHAR(255) COLLATE "C";
+ALTER TABLE "group_membership" ALTER COLUMN "group_id" SET DATA TYPE VARCHAR(255) COLLATE "C";
+ALTER TABLE "group_membership" ALTER COLUMN "member_id" SET DATA TYPE VARCHAR(255) COLLATE "C";
+ALTER TABLE "user_details" ALTER COLUMN "id" SET DATA TYPE VARCHAR(255) COLLATE "C";
```

Comment:
If the user id is not generated by us, there is the possibility, that it contains non-ASCII chars.

No sane implementation would do this, but we should be aware of this (very small) risk.